azurerm/resource_arm_dashboard.go

```diff
@@ -0,0 +1,158 @@
+package azurerm
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/portal/mgmt/2019-01-01-preview/portal"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDashboard() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDashboardCreateUpdate,
+		Read:   resourceArmDashboardRead,
+		Update: resourceArmDashboardCreateUpdate,
+		Delete: resourceArmDashboardDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateDashboardName,
+			},
+			"resource_group_name": azure.SchemaResourceGroupName(),
+			"location":            azure.SchemaLocation(),
+			"tags":                tags.Schema(),
+			"dashboard_properties": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Computed:  true,
+				StateFunc: normalizeJson,
+			},
+		},
+	}
+}
+
+func resourceArmDashboardCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).portal.DashboardsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	t := d.Get("tags").(map[string]interface{})
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	dashboardProps := d.Get("dashboard_properties").(string)
+
+	dashboard := portal.Dashboard{
+		Location: &location,
+		Tags:     tags.Expand(t),
+	}
+
+	var dashboardProperties portal.DashboardProperties
+
+	if err := json.Unmarshal([]byte(dashboardProps), &dashboardProperties); err != nil {
+		return fmt.Errorf("Error parsing JSON: %+v", err)
+	}
+	dashboard.DashboardProperties = &dashboardProperties
+
+	db, err := client.CreateOrUpdate(ctx, resourceGroup, name, dashboard)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Dashboard %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*db.ID)
```

Comment:
given the ID's used for `CreateOrUpdate` and `Get` can vary - can we set this ID using `d.SetID(*resp.ID)` below?